### PR TITLE
Warn when http URLs are configured

### LINKED ISSUE #14230: Explain that `configure_layout` URLs get embedded in iframes
### Motivation

The user does not know what to do when an iframe can't display the contents. 

Screenshot 2022-08-16 at 6 09 59 PM

### Pitch

A) The frontend can display a useful message instead of a blank page

B) Add a sentence to the docs that says https links should be preferred over http. And that the source can still prevent that the content be embedded, which is out of our control (I assume). 

### Alternatives

Raise an error or emit a warning. This would lead to false positives, as we can't (?) know if the iframe or external site will allow the url or not. Or maybe we can know it in advance? In this case, we should warn on the Python side too. 

### Additional context

[Discussed on Slack
](https://pytorch-lightning.slack.com/archives/CRBLFHY79/p1660651533273409)

cc @borda @rohitgr7 @krishnakalyan3 @hhsecond @williamFalcon 
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14230

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/lightning_app/utilities/layout.py [hunk 1]
```diff
@@ -79,11 +81,17 @@ def _collect_content_layout(layout: List[Dict], flow: "lightning_app.LightningFl
...
+            url = entry["content"]
+            if url.startswith("/"):
+                # The URL isn't fully defined yet. Looks something like ``self.work.url + /something``.
                 entry["target"] = ""
             else:
-                entry["target"] = entry["content"]
+                entry["target"] = url
+            if url.startswith("http") and not url.startswith("https") and is_running_in_cloud():
+                warnings.warn(
+                    f"You configured an http link {url[:32]}... but it may not be accessible in the cloud."
```
Comment (alecmerdler) [inline src/lightning_app/utilities/layout.py:92]: It will _never_ be accessible in the cloud due to `https`/`http` restrictions by modern browsers.

**Review by alecmerdler — APPROVED** [review]

File: src/lightning_app/utilities/layout.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_app/utilities/layout.py:92]: Got it, updated the text.

File: src/lightning_app/utilities/layout.py [hunk 2]
```diff
@@ -79,11 +81,17 @@ def _collect_content_layout(layout: List[Dict], flow: "lightning_app.LightningFl
...
-            # The URL isn't fully defined yet. Looks something like ``self.work.url + /something``.
-            if entry["content"].startswith("/"):
+            url = entry["content"]
+            if url.startswith("/"):
+                # The URL isn't fully defined yet. Looks something like ``self.work.url + /something``.
                 entry["target"] = ""
             else:
-                entry["target"] = entry["content"]
+                entry["target"] = url
+            if url.startswith("http") and not url.startswith("https") and is_running_in_cloud():
```
Comment (hhsecond) [inline src/lightning_app/utilities/layout.py:90]: nit: easier to just check for `url.startswith("http://")`

Comment (hhsecond) [inline src/lightning_app/utilities/layout.py:90]: nit
```suggestion
            if url.startswith("http://") and is_running_in_cloud():
```

**Review by hhsecond — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Comment by krishnakalyan3** [comment]
LGTM

**Review by manskx — APPROVED** [review]
